cmake/nuttx_toolchain.cmake

```diff
@@ -23,6 +23,13 @@
 # search. If the manual of the newly supported toolchain is different, you can
 # override these methods in the toolchain
 
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
```

Comment:
not sure why, but for me CMAKE_C_COMPILER_ID value is empty in this context

Reply:
OK, I know know. This implementation is wrong. `CMAKE_C_COMPILER_ID` is valid only after `project()` call

Reply:
> OK, I know now. This implementation is wrong. `CMAKE_C_COMPILER_ID` is valid only after `project()` call

Ok, you're right, now this bug is fixed, you can try this in you local